packages/cache/src/AzureBlobCacheStorage.ts

```diff
@@ -29,41 +31,65 @@ function createBlobClient(
 }
 
 export class AzureBlobCacheStorage extends CacheStorage {
-  constructor(private options: AzureBlobCacheStorageOptions) {
+  constructor(
+    private options: AzureBlobCacheStorageOptions,
+    private internalCacheFolder: string
+  ) {
     super();
   }
 
-  protected _fetch(hash: string, destinationFolder: string): Promise<boolean> {
-    const blobClient = createBlobClient(
-      this.options.connectionString,
-      this.options.container,
+  protected async _fetch(hash: string): Promise<string | undefined> {
+    const temporaryBlobOutputFolder = path.join(
+      this.internalCacheFolder,
+      "azure-blob",
       hash
     );
 
-    return blobClient
-      .download(0)
-      .then(response => {
-        const parentFolderWhereToExtractFolder = path.join(
-          destinationFolder,
-          ".."
+    if (!fs.existsSync(temporaryBlobOutputFolder)) {
```

Comment:
This method is deprecated, at least in NodeJS 12, the recommended way to test for a file/folder is to use fs.stats().
Not sure if we can say the same about "fs-extra" though

Reply:
Actually, the nodejs documentation says 
> `fs.exists()` is deprecated, but `fs.existsSync()` is not.

In any case, the function has not been deprecated from `fs-extra`, so we should be good.